frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -605,19 +605,20 @@ class SegmentsView extends React.Component<Props, State> {
         };
 
         const segmentsOfGroup = groupToSegmentsMap[group.groupId];
-        if (segmentsOfGroup == null) return;
```

Comment:
This is the root of all evil. Due to the early return, a group with no segments as direct children will not have its `visibilityEntry` entry  added to `newVisibleMap`. Thus the assumption that all child groups of a group already have an entry in `newVisibleMap` does not hold in lines 599-604 leading to the bug.